### PR TITLE
*: bump client-go with some regression tests

### LINKED ISSUE #56837: txn: memory of previous execution can not be released when retrying a transaction
## Enhancement
When retrying a transaction, for example, when we meet lock-conflict error, the memory of the previous execution can not be released until the txn commits successfully. It may cause conflict txn killed by tidb_mem_quota_query or make TiDB OOM

This issue was fix in https://github.com/tikv/client-go/pull/1451 since v8.4.0, while  I think we need PR to fix it in lower versions.

### LINKED ISSUE #57425: Slice bounds out of range after inserting many rows in a transaction
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
create table t1 (id int auto_increment key, b int, index(b));
begin;
insert into t1 (b) values (1),(2),(3),(4),(5),(6),(7),(8);
insert into t1 (b) select /*+ use_index(t1, b) */ id from t1;
insert into t1 (b) select /*+ use_index(t1, b) */ id from t1;
insert into t1 (b) select /*+ use_index(t1, b) */ id from t1;
insert into t1 (b) select /*+ use_index(t1, b) */ id from t1;
insert into t1 (b) select /*+ use_index(t1, b) */ id from t1;
insert into t1 (b) select /*+ use_index(t1, b) */ id from t1;
insert into t1 (b) select /*+ use_index(t1, b) */ id from t1;
insert into t1 (b) select /*+ use_index(t1, b) */ id from t1;
```

### 2. What did you expect to see? (Required)

Insert successfully.

### 3. What did you see instead (Required)

```
mysql> insert into t1 (b) select id from t1;
ERROR 1105 (HY000): runtime error: slice bounds out of range [-118:]
```

### 4. What is your TiDB version? (Required)

4fded5f4384a50762547d72c0c109ef0d33ddf1e

### PR DESCRIPTION
**Pull request by you06**
### What problem does this PR solve?

Issue Number: ref #55287, close #56837, ref #57338, close #57425

Problem Summary:

### What changed and how does it work?

Introduce the chunk comparison of common prefix (`amd64` and `arm64` only), which can accelerate a lot for indexes with very wide columns.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by ekexium — APPROVED** [review]

File: pkg/store/driver/txn/driver_test.go [hunk 1]
```diff
@@ -84,3 +84,7 @@ func TestWriteConflictPrettyFormat(t *testing.T) {
 		"key=‹›‹{metaKey=true, key=DB:56, field=TID:108}, originalKey=6d44423a3536000000fc00000000000000685449443a31303800fe, primary=›‹›‹{metaKey=true, key=DB:56, field=TID:108}, originalPrimaryKey=6d44423a3536000000fc00000000000000685449443a31303800fe›, reason=Optimistic " + kv.TxnRetryableMark
 	require.EqualError(t, newWriteConflictError(conflict), expectedStr)
 }
+
+func TestKeyValueOversize(t *testing.T) {
+	//key := make([]byte, math.MaxUint16+1)
```
Comment (cfzjywxk) [inline pkg/store/driver/txn/driver_test.go:89]: What is this?

File: pkg/executor/mem_reader.go [hunk 2]
```diff
@@ -882,6 +883,7 @@ func buildMemIndexMergeReader(ctx context.Context, us *UnionScanExec, indexMerge
 
 type memRowsIter interface {
 	Next() ([]types.Datum, error)
+	Close()
```
Comment (cfzjywxk) [inline pkg/executor/mem_reader.go:886]: Need comments to remind the necessity of calling Close`.

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**